examples/phase_estimator.py

```diff
@@ -0,0 +1,151 @@
+"""Creates and simulates a phase estimator circuit.
+
+=== EXAMPLE OUTPUT ===
+Estimation with 2qubits.
+Actual, Estimation (Raw binary)
+0.0000, 0.0000 (00)
+0.1000, 0.0000 (00)
+0.2000, 0.2500 (01)
+0.3000, 0.2500 (01)
+0.4000, 0.5000 (10)
+0.5000, 0.5000 (10)
+0.6000, 0.5000 (10)
+0.7000, 0.7500 (11)
+0.8000, 0.7500 (11)
+0.9000, 0.0000 (00)
+RMS Error: 0.2915
+
+Estimation with 4qubits.
+Actual, Estimation (Raw binary)
+0.0000, 0.0000 (0000)
+0.1000, 0.1250 (0010)
+0.2000, 0.1875 (0011)
+0.3000, 0.3125 (0101)
+0.4000, 0.3750 (0110)
+0.5000, 0.5000 (1000)
+0.6000, 0.6250 (1010)
+0.7000, 0.6875 (1011)
+0.8000, 0.8125 (1101)
+0.9000, 0.8750 (1110)
+RMS Error: 0.0177
+
+Estimation with 8qubits.
+Actual, Estimation (Raw binary)
+0.0000, 0.0000 (00000000)
+0.1000, 0.1016 (00011010)
+0.2000, 0.1992 (00110011)
+0.3000, 0.3008 (01001101)
+0.4000, 0.3984 (01100110)
+0.5000, 0.5000 (10000000)
+0.6000, 0.6016 (10011010)
+0.7000, 0.6992 (10110011)
+0.8000, 0.8008 (11001101)
+0.9000, 0.8984 (11100110)
+RMS Error: 0.0011
+"""
+
+
+import numpy as np
+import cirq
+
+
+def phase_op(phi):
+    """An example unitary 1-qubit operation U with an eigen vector |0>
+    and an eigen value exp(2*Pi*i*phi)
+    """
+
+    gate = cirq.SingleQubitMatrixGate(
+        matrix=np.array([[np.exp(2*np.pi*1.0j*phi), 0], [0, 1]]))
+    return gate
+
+
+class QftInverse(cirq.Gate, cirq.CompositeGate):
+    """Quantum gate for the inverse Quantum Fourier Transformation
+    """
+
+    def default_decompose(self, qubits):
+        """A quantum circuit with the following structure.
+
+        ---H----R_2---R_3---R_4--------------------------------------
```

Comment:
Use the diagram that actually gets printed out.

```
In [5]: cirq.Circuit.from_ops(QftInverse().default_decompose(cirq.LineQubit.range(4)))
Out[5]: 
0: ───H───@────────@─────────@──────────────────────────────────────────────────
          │        │         │
1: ───────@^-0.5───┼─────────┼──────────H───@────────@──────────────────────────
                   │         │              │        │
2: ────────────────@^-0.25───┼──────────────@^-0.5───┼─────────H───@────────────
                             │                       │             │
3: ──────────────────────────@^-0.125────────────────@^-0.25───────@^-0.5───H───
```

Reply:
I got it. I will update it.